controllers/machinehealthcheck_controller_test.go

```diff
@@ -561,21 +559,33 @@ func TestClusterToMachineHealthCheck(t *testing.T) {
 	defer close(doneMgr)
 
 	// END: setup test environment
+	namespace := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{GenerateName: "mhc-test-"}}
+	g.Expect(r.Client.Create(context.Background(), namespace)).To(Succeed())
+	defer func() {
+		g.Expect(r.Client.Delete(context.Background(), namespace))
+	}()
+	namespaceName := namespace.Name
+
+	otherNamespace := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{GenerateName: "mhc-test-"}}
+	g.Expect(r.Client.Create(context.Background(), otherNamespace)).To(Succeed())
+	defer func() {
+		g.Expect(r.Client.Delete(context.Background(), otherNamespace))
+	}()
+	otherNamespaceName := otherNamespace.Name
```

Comment:
This is needed because previously envtest was only allowing this due to disabling of admission plugins (namespace not existing already).